packages/components/src/tools-panel/tools-panel-header/component.tsx

```diff
@@ -199,18 +200,23 @@ const ToolsPanelHeader = (
 						isSmall: true,
 						describedBy: dropdownMenuDescriptionText,
 					} }
+					popoverProps={ dropdownMenuProps }
```

Comment:
If we're passing `dropdownMenuProps`, then they should be applied to the `DropdownMenu` component, and not just to its `popoverProps` prop.

Something like:

```suggestion
					{ ...dropdownMenuProps }
```

The question then becomes about what prop takes the precedence? Should values from `dropdownMenuProps` override the props normally passed to `DropdownMenu` ? Should we merge values in `menuProps` and `toggleProps` ? @aaronrobertshaw  and @talldan what do you think?

Reply:
Gotcha. I wasn't able to get it to function properly using the `...dropdownMenuProps` like that, as-is. 

![CleanShot 2023-11-09 at 15 00 23](https://github.com/WordPress/gutenberg/assets/1813435/2ce4a828-718f-4055-88aa-05d26deeb1f4)

Reply:
>  I wasn't able to get it to function properly using the ...dropdownMenuProps like that, as-is.

The missing piece to adopt the `{ ...dropdownMenuProps }` approach is that the props being passed to the component also need a small change. I've included a diff to illustrate in my [review](https://github.com/WordPress/gutenberg/pull/55785#pullrequestreview-1723584367).

> @aaronrobertshaw and @talldan what do you think?

The experience for the ToolsPanel is fairly curated. That makes me think we don't want the new `dropdownMenuProps` overriding what the component is doing. Merging might be a compromise option. Do we actually have a use case that requires the extra control yet?